utils/data.py

```
@@ -0,0 +1,73 @@
+    파일이 없거나 손상된 경우 빈 리스트 반환.
+    """
+    file_path = DATA_DIR / filename
+
+    if not file_path.exists():
+        return []
+
+    try:
+        with open(file_path, "r", encoding="utf-8") as f:
+            return json.load(f)
+    except json.JSONDecodeError as e:
+        logger.error(f"JSON 파싱 에러 ({filename}): {e}")
```

Ruff (0.14.13)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)